infra/cifuzz/filestore/gitlab/__init__.py

```diff
@@ -0,0 +1,121 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Empty filestore implementation for platforms that haven't implemented it."""
+import logging
+
+import shutil
+import os
+import json
+import tempfile
+
+import environment
+import http_utils
+import filestore
+
+# pylint:disable=no-self-use,unused-argument
+
+
+class GitlabFilestore(filestore.BaseFilestore):
```

Comment:
I guess we could have another filestore which would be a gitlab repository instead of gitlab's artifacts

Reply:
Actually this used to be my approach because of the limitations of GitLab's artifacts API. The existing git filestore covers many things already and only relies on `upload_crashes()`, `upload_build()`, and `download_build()` to be implemented by the CI filestore.

Reply:
@jonathanmetzman would you like 2 gitlab filestores ? one with repository inheriting from the git filestore, and another one with cache/artifacts ?

Reply:
You don't need to implement another filestore.
Just use the git filestore and pass in the gitlab filestore as the `ci_filestore` argument like so: https://github.com/google/oss-fuzz/blob/master/infra/cifuzz/filestore/git/__init__.py#L56